src/interrupt/idt.rs

```diff
@@ -0,0 +1,73 @@
+use core::mem::size_of;
+
+use crate::{interrupt::exception::CpuException, sync::singleton::Singleton};
+
+use super::{
+	handler::{divide_error::divide_error_handler, page_fault::page_fault_handler},
+	idte::IDTE,
+};
+
+const IDTE_COUNT: usize = 256;
+
+#[repr(align(8))]
+pub struct IDT {
+	entry: [IDTE; IDTE_COUNT],
+}
+
+impl IDT {
+	pub fn init() {
+		let de = IDTE::interrupt_kernel(divide_error_handler as usize);
+		let pf = IDTE::interrupt_kernel(page_fault_handler as usize);
+
+		IDT.lock().write_exception(CpuException::DE, de);
+		IDT.lock().write_exception(CpuException::PF, pf);
+
+		IDT.lock().load();
+	}
+
+	pub const fn new() -> Self {
+		Self {
+			entry: [IDTE::null(); IDTE_COUNT],
+		}
+	}
+
+	pub fn write_exception(&mut self, e: CpuException, entry: IDTE) {
+		if e == CpuException::Reserved {
+			panic!("idt: don't use the exception reserved for cpu.");
+		}
+		self.entry[e as usize] = entry
+	}
+
+	pub fn write_interrupt(&mut self, index: usize, entry: IDTE) {
+		if index >= IDTE_COUNT {
```

Comment:
```suggestion
		if  index < 32 ||  index >= IDTE_COUNT {
```
32보다 낮은 인터럽트 엔트리 등록 방지